Code/Framework/AzToolsFramework/AzToolsFramework/ViewportSelection/EditorTransformComponentSelection.cpp

```diff
@@ -3623,6 +3626,22 @@ namespace AzToolsFramework
                 m_selectedEntityIds.erase(focusRoot);
             }
         }
+
+        // Do not create manipulators for any entities marked as read only
+        if (auto readOnlyEntityPublicInterface = AZ::Interface<ReadOnlyEntityPublicInterface>::Get())
+        {
+            for (auto it = m_selectedEntityIds.begin(); it != m_selectedEntityIds.end();)
+            {
+                if (readOnlyEntityPublicInterface->IsReadOnly(*it))
+                {
+                    it = m_selectedEntityIds.erase(it);
+                }
+                else
+                {
+                    ++it;
+                }
+            }
```

Comment:
question: Do we have a utility function to do this already? Rather than using this directly here it might be nice to write a function to do this `erase_if` logic on the `set` (as we can't use the nice `erase/remove` idiom here). This will keep the application logic higher level and will be useful for future use too (aaaand we can then write a test for it! 🙂)

Reply:
Cool we do actually have `AZStd::erase_if` already, and it works here!  I will update :)  Thanks for the heads up

Reply:
Oh awesome! Good find! 😄